app/controllers/users_controller.rb

```diff
@@ -11,7 +11,7 @@ def register
 
   def login
     user = User.find_by(email_id: user_params[:email_id])
-    if user.nil? || !user.authenticate(user_params[:password])
+    if user.nil? || user.deleted? || !user.authenticate(user_params[:password])
```

Comment:
If the user is temporarily deleted, allow them to login and restore the account. 

If the user is permanently deleted, dont allow them to login.